### PR TITLE
CONTRIBUTING: Add licensing details (#1994)

### LINKED ISSUE #1994: CONTRIBUTING: Make it clear not to copy/reference GPL code
I think there should be a section in `CONTRIBUTING.md` stating that contributors are not allowed to use GPL code (in particular, the GNU coreutils source code) as a reference. That section should probably also give a quick explanation of why this is the case. Stating this policy explicitly is probably also safer for the project in general. It might also be helpful to add licenses which we definitely can reference and licenses which we cannot. Something like:
```
You are only allowed to reference code which is compatible with the MIT License, examples include:
- MIT
- BSD
- CC0
- Unlicense

You are **not** allowed to reference code with following licenses:
- GPL
- LGPL
- No license specified
- And many more
```

### PR DESCRIPTION
**Pull request by dhodder**
Document existing practices around licensing, including making it clear
not to copy/reference GPL code.

Intended to resolve #1994.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,6 +25,9 @@ search the issues to make sure no one else is working on it.
 1. Don't hesitate to move common functions into uucore if they can be reused by other binaries.
 1. Unsafe code should be documented with Safety comments.
 
+uutils is original code. It cannot contain code from existing GNU or Unix-like
```
Comment (tertsdiepraam) [inline CONTRIBUTING.md:28]: Could you add this to the list instead of a separate paragraph?

**Review by tertsdiepraam — COMMENTED** [review]
Great! I have one more question (maybe for @sylvestre): what should we do with Apache 2.0 only dependencies?

**Comment by sylvestre** [comment]
great indeed! thanks for paying attention to licenses!

File: CONTRIBUTING.md (hunk 1 above)
Comment (dhodder) [reply CONTRIBUTING.md:28]: Sure!

**Comment by dhodder** [comment]
> Great! I have one more question (maybe for @sylvestre): should we allow Apache 2.0 only dependencies?

This question has been asked in issue #2086. I don't know what the consensus is, it may even be a nuanced answer based on the dependency rather than a straight out yes or no. (See for example, the [licensing policy of FreeBSD](https://www.freebsd.org/internal/software-license/): they use BSD/ISC/MIT licensing but their core team allows Apache on a case-by-case basis.)

**Comment by tertsdiepraam** [comment]
Issue #2086 was indeed why I brought it up. The main difference with MIT seems to be that you have to state the changes you make, so I'm leaning towards allowing it for dependencies, but not allowing copying code from Apache 2.0 projects. This ensures that we don't modify the code without stating the changes. Does that make sense?

**Comment by dhodder** [comment]
> Issue #2086 was indeed why I brought it up. The main difference with MIT seems to be that you have to state the changes you make, so I'm leaning towards allowing it for dependencies, but not allowing copying code from Apache 2.0 projects. This ensures that we don't modify the code without stating the changes. Does that make sense?

That makes senses to me. We could add a third license category, between "acceptable" and "licenses we will not use".

* "Licenses accepted for references, on a case-by-cases basis," or maybe
* "Licenses accepted in referenced dependencies, when there is no obvious MIT alternative"?

**Comment by tertsdiepraam** [comment]
Sounds good, but I'd like to hear what @sylvestre and @rivy think before we commit to that.

**Comment by sylvestre** [comment]
Sounds great! thanks :)

**Comment by dhodder** [comment]
Thank you @sylvestre and @tertsdiepraam .

@rivy , do you have any opinions please. Should we add a third license category to `CONTRIBUTING.md` between "acceptable" and "licenses we will not use":

>Licenses accepted in referenced dependencies, on a case-by-case basis, when there is no MIT-licensed alternative:
>
>* Apache License, version 2.0

Thanks.

**Pull request merged**